kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQueryRecordConverter.java

```diff
@@ -73,13 +67,53 @@ public BigQueryRecordConverter(boolean shouldConvertDoubleSpecial) {
    */
   public Map<String, Object> convertRecord(SinkRecord kafkaConnectRecord) {
     Schema kafkaConnectSchema = kafkaConnectRecord.valueSchema();
+    Object kafkaConnectValue = kafkaConnectRecord.value();
+    if (kafkaConnectSchema == null) {
+      if (kafkaConnectValue instanceof Map) {
+        return (Map<String, Object>) validateSchemalessRecord(kafkaConnectValue);
+      }
+      throw new ConversionConnectException("Only Map objects supported in absence of schema for " +
+              "record conversion to BigQuery format.");
+    }
     if (kafkaConnectSchema.type() != Schema.Type.STRUCT) {
       throw new
           ConversionConnectException("Top-level Kafka Connect schema must be of type 'struct'");
     }
     return convertStruct(kafkaConnectRecord.value(), kafkaConnectSchema);
   }
 
+  private Object validateSchemalessRecord(Object value) {
```

Comment:
`convertSchemalessRecord` would be more explicit, since some conversion can happen.

If we want to optimize for the case where the record is clean and does not require any conversion, then we could do a two-pass approach to avoid deep-copying every single record: validate first without copying, and then convert if necessary. This will avoid putting too much pressure on the garbage-collector and reduce unnecessary memory allocations.

Reply:
given that we already deep-copy records in the case when we do have a schema, I'd be ok if we leave this optimization for a follow-up PR if necessary